### PR TITLE
Overlay Vulnerability

### PR DESCRIPTION
**Pull request by ErebusZ**

### CODE REVIEW

File: MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/meta.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "risk_rating": "high",
+    "short_description": "Overlay Vulnerability is a type of vulnerability where an attacker can overlay malicious content on top of legitimate content, leading to potential security breaches.",
+    "references": {
+        "Overlay Attacks: Top Techniques And How To Counter Them": "https://www.appsealing.com/overlay-attacks/",
+        "Mobile overlay attacks on Android": "https://www.ikarussecurity.com/en/mobile-device-management-en/mobile-overlay-attacks-on-android/"
+    },
+    "title": "Overlay Vulnerability",
+    "privacy_issue": false,
```
Comment (BlueSquare1) [inline MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/meta.json:9]:
```suggestion
    "privacy_issue": true,
```

File: MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/recommendation.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+# Overlay Vulnerability
+
+To mitigate overlay vulnerabilities in mobile applications, developers should implement robust security measures with a focus on enhancing settings and user interactions. First and foremost, ensure that user inputs and application settings undergo thorough validation and filtering to prevent potential exploitation by malicious overlays. Pay special attention to sensitive settings that can be manipulated by overlays, such as authentication parameters or access controls. Implement strict input validation checks on user-modifiable settings to prevent unauthorized changes. When handling settings related to user authentication or authorization, enforce strong authentication mechanisms and authorization controls to minimize the impact of potential overlay attacks. Provide clear user guidance within the application on recognizing and avoiding suspicious overlays. Regularly update the application to address any newly identified vulnerabilities, and consider implementing overlay detection mechanisms to enhance the application's resilience against potential threats.
```
Comment (BlueSquare1) [inline MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/recommendation.md:4]: Separate points would be better than one giant paragraph

File: MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/meta.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+    "risk_rating": "high",
+    "short_description": "Overlay Vulnerability is a type of vulnerability where an attacker can overlay malicious content on top of legitimate content, leading to potential security breaches.",
+    "references": {
+        "Overlay Attacks: Top Techniques And How To Counter Them": "https://www.appsealing.com/overlay-attacks/",
+        "Mobile overlay attacks on Android": "https://www.ikarussecurity.com/en/mobile-device-management-en/mobile-overlay-attacks-on-android/"
+    },
+    "title": "Overlay Vulnerability",
```
Comment (BlueSquare1) [inline MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/meta.json:8]: I think Tapjacking is the technical name for this

File: MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/recommendation.md (hunk 2 above)
Comment (BlueSquare1) [reply MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/recommendation.md:4]: You can mention this:

Newer versions of Android have introduced several mitigations, including the following:

- On Android 6 (API level 23) and higher, users have to explicitly grant the permission for the app to create an overlay window.
- On Android 12 (API level 31) and higher, apps can pass true into [Window.setHideOverlayWindows()](https://developer.android.com/reference/android/view/Window#setHideOverlayWindows(boolean)).

**Review by BlueSquare1 — DISMISSED** [review]

File: MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/recommendation.md (hunk 2 above)
Comment (ErebusZ) [reply MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/recommendation.md:4]: noted

File: MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/meta.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    "privacy_issue": true,
+    "security_issue": true,
+    "categories": {
+        "OWASP_MASVS_L1": [
+            "MSTG_PLATFORM_2"
+        ],
+        "OWASP_MASVS_L2": [
+            "MSTG_PLATFORM_2"
+        ]
```
Comment (elyousfi5) [inline MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/meta.json:17]: I think these should can be added too: 

`"PCI_STANDARDS":[
      "REQ_2_2",
      "REQ_6_2",
"REQ_6_3"
    ]`

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+# Tapjacking Vulnerability
+
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md:3]: Remove the title.

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+
+# Overlay Vulnerability
+
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md:3]: Please remove.

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+The Tapjacking Vulnerability is a security flaw that allows an attacker to overlay malicious content or interfaces on top of legitimate applications, tricking users into interacting with the attacker's content instead. This can lead to various malicious activities, such as stealing sensitive information, capturing login credentials, or performing unauthorized actions on the user's behalf.
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md:4]: Show sample of vulnerable code in the resources XML and in Kotlin and Java.

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+To mitigate Tapjacking vulnerabilities in mobile applications, developers should:
+
+ * Implement robust security measures for mobile applications to mitigate overlay vulnerabilities.
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md:6]: This has nothing to do with the finding we are reporting, we are just ensuring that a certain setting should not be set.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+ * Thoroughly validate and filter user inputs and application settings to prevent exploitation by malicious overlays.
+ * Pay special attention to sensitive settings vulnerable to manipulation by overlays, such as authentication parameters or access controls.
+ * Implement strict input validation checks on user-modifiable settings to prevent unauthorized changes.
+ * Provide clear user guidance within the application on recognizing and avoiding suspicious overlays.
+ * Regularly update the application to address newly identified vulnerabilities.
+ * Consider implementing overlay detection mechanisms to enhance the application's resilience against potential threats.
+ * Set the filterTouchesWhenObscured on views to true.
+ * Set security policies for touch event using onFilterTouchEventForSecurity.
+ * On Android 6 (API level 23) and higher, users have to explicitly grant the permission for the app to create an overlay window.
+ * On Android 12 (API level 31) and higher, apps can pass true into Window.setHideOverlayWindows().
```
Comment (BlueSquare1) [inline MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md:13]: These mitigations are in the OS itself and not something the user should do, we should just mention them.

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
+
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md:1]: Remove empty line.

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
+
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md:1]: Remove empty line.

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+To mitigate Tapjacking vulnerabilities in mobile applications, developers should:
+
+ * Thoroughly validate and filter user inputs and application settings to prevent exploitation by malicious overlays.
+ * Pay special attention to sensitive settings vulnerable to manipulation by overlays, such as authentication parameters or access controls.
+ * Implement strict input validation checks on user-modifiable settings to prevent unauthorized changes.
+ * Provide clear user guidance within the application on recognizing and avoiding suspicious overlays.
+ * Regularly update the application to address newly identified vulnerabilities.
+ * Consider implementing overlay detection mechanisms to enhance the application's resilience against potential threats.
+ * Set the filterTouchesWhenObscured on views to true.
+ * Set security policies for touch event using onFilterTouchEventForSecurity.
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md:11]:
```suggestion
 * Set the `filterTouchesWhenObscured` on views to true.
 * Set security policies for touch event using `onFilterTouchEventForSecurity`.
```

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_your_layout);
+
+        Button yourButton = findViewById(R.id.yourButtonId);
+        yourButton.filterTouchesWhenObscured = true
+        yourButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                // Handle button click
+            }
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md:49]: Why is this secure?

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+class YourActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_your_layout)
+
+        val yourButton: Button = findViewById(R.id.yourButtonId)
+        yourButton.setOnClickListener {
+            // Handle button click
+        }
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md:57]: Why is this insecure? I don't understand the difference the secure code?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md (hunk 14 above)
Comment (ErebusZ) [reply MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md:57]: the diffrence is that the secure code has the `filterTouchesWhenObscured` set to true

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md (hunk 10 above)
Comment (3asm) [reply MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/description.md:1]: Ping.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md (hunk 13 above)
Comment (ErebusZ) [reply MOBILE_CLIENT/COMMON/_HIGH/TAPJACKING_VULNERABILITY/recommendation.md:49]: the diffrence is that the secure code has the filterTouchesWhenObscured set to true

**Review by 3asm — APPROVED** [review]

**Pull request merged**